samples/SampleApp/Startup.cs

```diff
@@ -61,9 +61,11 @@ public static void Main(string[] args)
 
                     // The following section should be used to demo sockets
                     //options.ListenUnixSocket("/tmp/kestrel-test.sock");
-
+                })
+                .UseLibuv(options =>
```

Comment:
That just means that hosting services don't get injected into the ITransportFactory for free. For example, `LibuvTransportFactory` depends on `IOptions<LibuvTransportOptions>, `IApplicationLifetime` and `ILoggerFactory`.

Do you want to use ActivatorUtilities to activate a registered ITransportFactory type on KestrelServerOptions? That seems really gross.